pkg/nsx/services/securitypolicy/firewall.go

```diff
@@ -127,11 +139,70 @@ func (service *SecurityPolicyService) CreateOrUpdateSecurityPolicy(obj *v1alpha1
 	// WrapHighLevelSecurityPolicy will modify the input security policy, so we need to make a copy for the following store update.
 	finalSecurityPolicyCopy := *finalSecurityPolicy
 	finalSecurityPolicyCopy.Rules = finalSecurityPolicy.Rules
-	infraSecurityPolicy, err := service.WrapHierarchySecurityPolicy(finalSecurityPolicy, finalGroups)
-	if err != nil {
-		return err
+
+	if isVpcEnabled(service) {
```

Comment:
Here, as discussed, whether we should add more checks of t1 or vpc nsxt version?

Reply:
we will hand this vpc network activation check with nex version during nsx operator startup. So, we don't need to handle it here.